tmail-backend/blob/secondary-blob-store/src/main/java/com/linagora/tmail/blob/secondaryblobstore/FailedBlobEvents.java

```diff
@@ -0,0 +1,104 @@
+package com.linagora.tmail.blob.secondaryblobstore;
+
+import java.util.Collection;
+
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.BucketName;
+import org.apache.james.core.Username;
+import org.apache.james.events.Event;
+
+public interface FailedBlobEvents {
+    interface BlobEvent extends Event {
+        Username USERNAME = Username.of("SecondaryBlobStore");
+
+        ObjectStorageIdentity getFailedObjectStorage();
+    }
+
+    record BlobAddition(EventId eventId, BucketName bucketName, BlobId blobId, ObjectStorageIdentity failedObjectStorage) implements BlobEvent {
+        @Override
+        public Username getUsername() {
+            return USERNAME;
+        }
+
+        @Override
+        public boolean isNoop() {
+            return false;
+        }
+
+        @Override
+        public EventId getEventId() {
+            return eventId;
+        }
+
+        @Override
+        public ObjectStorageIdentity getFailedObjectStorage() {
+            return failedObjectStorage;
+        }
+    }
+
+    record BlobDeletion(EventId eventId, BucketName bucketName, BlobId blobId, ObjectStorageIdentity failedObjectStorage) implements BlobEvent {
```

Comment:
Is that record necessary? For the deletion of one blobId we could just call the record below with `ImmutableList.of(blobId)` or soemthing like that, right?